src/App.js

```diff
@@ -1,18 +1,22 @@
 import React, { Component } from 'react';
-import logo from './logo.svg';
-import './App.css';
+import {
+  Route
+} from 'react-router-dom';
+
+import MainHeader from './components/MainHeader';
+
+import ViewHome from './views/Home';
+import ViewFavorites from './views/Favorites';
+import ViewDetails from './views/Details';
 
 class App extends Component {
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      <div>
+        <MainHeader />
+        <Route path="/gif/:gifId" component={ViewDetails} />
+        <Route exact path="/your-favorites" component={ViewFavorites} />
```

Comment:
no need of exact here since there is no other possible conflict to 'your-favorites'

Reply:
I had an odd behavior without the exacts in this case. Without exact, the react-router navlink component apply the active class in all the componets, so that's why I let the exacts :/